### PR TITLE
vochain: also use our new LRU cache for envelope heights

### PR DESCRIPTION
**Pull request by mvdan**
(see commit message)

### CODE REVIEW

**Comment by mvdan** [comment]
@p4u please test this because I don't think `go test ./...` is covering these caches very well, and I could have made a mistake.

File: vochain/scrutinizer/vote.go [hunk 1]
```diff
@@ -209,24 +209,33 @@ func (s *Scrutinizer) GetEnvelopes(processId []byte, max, from int,
...
+			count, err := s.db.Count(&indexertypes.VoteReference{},
+				badgerhold.Where("ProcessID").Eq(processID).Index("ProcessID"))
+			if err != nil {
+				return err
+			}
+			return uint64(count)
+
+		})
+		switch val := val.(type) {
+		case error:
```
Comment (NateWilliams2) [inline vochain/scrutinizer/vote.go]: In what case could GetOrUpdate return an error?

**Review by NateWilliams2 — COMMENTED** [review]
Looks fine to me, although I'm confused why in the case of envelopes you check the type of the returned interface, and in the case of blocks you simply cast it.

File: vochain/scrutinizer/vote.go (hunk 1 above)
Comment (mvdan) [reply vochain/scrutinizer/vote.go]: Thanks for flagging this - it's indeed a bit tricky, and it's on me to not have documented it better. It's now documented.

**Comment by mvdan** [comment]
I've documented why the cached height count can be an error. The block cache can never contain errors because the way we build the values (GetBlockByHeight, i.e. `app.Node.BlockStore().LoadBlock`) can never error.

Arguably this could be a bug on the Tendermint side, as it sounds like an API that should return an error, but it doesn't. It seems to panic if something goes very wrong.

**Comment by p4u** [comment]
Looks good to me, pretty smart. I have tested with dockersuit but for deeply test it we need to deploy it to the development network and run some benchmarking. I'll merge.

**Pull request merged**